Makefile

```diff
@@ -138,6 +111,78 @@ auto_usera.o: \
 compile auto_usera.c
 	./compile auto_usera.c
 
+auto_userd.c: \
+auto-str conf-users
+	./auto-str auto_userd `head -2 conf-users | tail -1` > auto_userd.c
+
+auto_userd.o: \
+compile auto_userd.c
+	./compile auto_userd.c
+
+auto_userl.c: \
+auto-str conf-users
+	./auto-str auto_userl `head -3 conf-users | tail -1` > auto_userl.c
+
+auto_userl.o: \
+compile auto_userl.c
+	./compile auto_userl.c
+
+auto_usero.c: \
+auto-str conf-users
+	./auto-str auto_usero `head -4 conf-users | tail -1` > auto_usero.c
+
+auto_usero.o: \
+compile auto_usero.c
+	./compile auto_usero.c
+
+auto_userp.c: \
+auto-str conf-users
+	./auto-str auto_userp `head -5 conf-users | tail -1` > auto_userp.c
+
+auto_userp.o: \
+compile auto_userp.c
+	./compile auto_userp.c
+
+auto_userq.c: \
+auto-str conf-users
+	./auto-str auto_userq `head -6 conf-users | tail -1` > auto_userq.c
+
+auto_userq.o: \
+compile auto_userq.c
+	./compile auto_userq.c
+
+auto_userr.c: \
+auto-str conf-users
+	./auto-str auto_userr `head -7 conf-users | tail -1` > auto_userr.c
+
+auto_userr.o: \
+compile auto_userr.c
+	./compile auto_userr.c
+
+auto_users.c: \
+auto-str conf-users
+	./auto-str auto_users `head -8 conf-users | tail -1` > auto_users.c
+
+auto_users.o: \
+compile auto_users.c
+	./compile auto_users.c
+
+auto_groupn.c: \
+auto-str conf-groups
+	./auto-str auto_groupn `head -1 conf-groups | tail -1` > auto_groupn.c
+
+auto_groupn.o: \
+compile auto_groupn.c
+	./compile auto_groupn.c
+
+auto_groupq.c: \
+auto-str conf-groups
+	./auto-str auto_groupq `head -2 conf-groups` > auto_groupq.c
```

Comment:
Missing a `| tail -1` again. I’ve always thought it was error-prone that we don’t do `| tail -1` in the `head -1` case. Maybe that would have helped here?

Reply:
Also, I think `groupq` is supposed to be the first line of `conf-groups` and `groupn` the second?

Reply:
I did have that wrong.  It's now fixed.  By hand since this part has been such trouble for me:

Here is the contents of `conf-groups`:

    qmail
    nofiles

    These are the qmail groups. The second group should not have access to
    any files, but it must be usable for processes; this requirement
    excludes the ``nogroup'' and ``nobody'' groups on many systems.

auto_groupq is the first entry, with auto_groupn the second entry.  The tail -1 call needs to be on the second entry in conf-groups, but alphabetically the first group (`n`, not `q`).

    $ head -1 conf-groups
    qmail

    $ head -2 conf-groups | tail -1
    nofiles

Such that the following C files are produced:

    ./auto-str auto_groupn `head -2 conf-groups | tail -1`
    char auto_groupn[] = "\
    \156\157\146\151\154\145\163\
    ";

    ./auto-str auto_groupq `head -1 conf-groups` > auto_groupq.c
    char auto_groupq[] = "\
    \161\155\141\151\154\
    ";

Where `\156` is is `n` and `\161` is `q`.